src/save.rs

```diff
@@ -0,0 +1,43 @@
+use anyhow::{format_err, Result};
```

Comment:
The root directory is getting a bit crowded. Perhaps move this to a subfolder? (named: 'io' or 'disk_io' ?).

What do you think? I'm to new to rust to argue for what's common project structuring.

Adding "logging" is in the list of issues which also pertains to disk storage so those files will probably share this folder.

Reply:
Yeah root is definitely getting crowded. I assume you're more of a fan of the folder/ + mod.rs in folder vs folder.rs + folder/ layout then? Otherwise root is just as crowded :P

Reply:
I’ve only used the mod.rs variant so far. I find it’s quite a clean structure